controllers/collector_controller.go

```diff
@@ -132,28 +136,29 @@ func (r *CollectorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return r.setCollectorActive(ctx, col, nil)
 	}
 
+	workerContext, cancel := context.WithCancel(ctx)
+	collectorWorker = &CollectorWorker{
```

Comment:
I'm confused why this got moved outside of the type check for `StartableCollector` - I don't think we want a dedicated worker for every collector.  Could you help me understand this change?